source/Halibut.Tests/PollingServiceTimeoutsFixture.cs

```diff
@@ -0,0 +1,113 @@
+                       .As<LatestClientAndLatestServiceBuilder>()
+                       .NoService()
+                       .WithStandardServices()
+                       .WithHalibutTimeoutsAndLimits(halibutTimeoutsAndLimits)
+                       .Build(CancellationToken))
+            {
+                var client = clientAndService.CreateAsyncClient<IEchoService, IAsyncClientEchoServiceWithOptions>();
+
+                var stopwatch = Stopwatch.StartNew();
+                (await AssertException.Throws<HalibutClientException>(async () => await client.SayHelloAsync("Hello", new(CancellationToken, CancellationToken.None))))
+                    .And.Message.Should().Contain("A request was sent to a polling endpoint, but the polling endpoint did not collect the request within the allowed time (00:00:05), so the request timed out.");
+                stopwatch.Stop();
+
+                stopwatch.Elapsed.Should().BeLessThan(TimeSpan.FromSeconds(8), "Should have timed out quickly");
+            }
+        }
+
+        [Test]
+        [LatestClientAndLatestServiceTestCases(testNetworkConditions: false, testListening: false)]
+        public async Task WhenThePollingRequestQueueTimeoutIsReached_ButTheResponseIsReceivedBeforeThePollingRequestMaximumMessageProcessingTimeoutIsReached_TheRequestShouldSucceed(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            var halibutTimeoutsAndLimits = new HalibutTimeoutsAndLimitsForTestsBuilder().Build();
+            halibutTimeoutsAndLimits.PollingRequestQueueTimeout = TimeSpan.FromSeconds(5);
+            halibutTimeoutsAndLimits.PollingRequestMaximumMessageProcessingTimeout = TimeSpan.FromSeconds(100);
+
+            var responseDelay = TimeSpan.FromSeconds(10);
+
+            await using (var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
+                             .As<LatestClientAndLatestServiceBuilder>()
+                             .WithDoSomeActionService(() => Thread.Sleep(responseDelay))
+                             .WithHalibutTimeoutsAndLimits(halibutTimeoutsAndLimits)
+                             .WithInstantReconnectPollingRetryPolicy()
+                             .Build(CancellationToken))
+            {
+                var doSomeActionClient = clientAndService.CreateAsyncClient<IDoSomeActionService, IAsyncClientDoSomeActionServiceWithOptions>();
+
+                var stopwatch = Stopwatch.StartNew();
+                await doSomeActionClient.ActionAsync(new(CancellationToken, CancellationToken.None));
+                stopwatch.Stop();
+
+                stopwatch.Elapsed.Should()
+                    .BeGreaterThan(halibutTimeoutsAndLimits.PollingRequestQueueTimeout, "Should have waited longer than the PollingRequestQueueTimeout").And
+                    .BeLessThan(responseDelay + TimeSpan.FromSeconds(5), "Should have received the response after the 10 second delay + 5 second buffer");
```

Comment:
nit: Is this last assertion necessary? If it took 15.1s on a busy build server would that still be ok since the call completed successfully and it exceeds the `PollingRequestQueueTimeout`

Reply:
No it's redundant isn't it

Reply:
Removed